executor/load_data.go

```diff
@@ -112,7 +112,12 @@ type LoadDataInfo struct {
 // SetMaxRowsInBatch sets the max number of rows to insert in a batch.
 func (e *LoadDataInfo) SetMaxRowsInBatch(limit uint64) {
 	e.maxRowsInBatch = limit
-	e.rows = make([][]types.Datum, 0, limit)
+	if uint64(cap(e.rows)) < limit {
+		e.rows = make([][]types.Datum, 0, limit)
+		for i := 0; uint64(i) < limit; i++ {
+			e.rows = append(e.rows, make([]types.Datum, len(e.Table.Cols())))
```

Comment:
maybe we only need new `limit - cap(e.rows)` types.Datum here if the limit can be dynamical change 

and the other optimize maybe or not work is `make([]types.Datum, (limit -cap(e.rows)) * len(e.Table.Cols()))` once and then cut slice from this big array to append

Reply:
seems trace not support load data now ?

Reply:
@lysu currently init will be called only once. in another pr making variable settable will do this